clang/lib/Sema/SemaBounds.cpp

```diff
@@ -532,39 +532,60 @@ namespace {
 }
 
 namespace {
-  class FindMemberHelper : public RecursiveASTVisitor<FindMemberHelper> {
+  class FindLValueHelper : public RecursiveASTVisitor<FindLValueHelper> {
     private:
       Sema &SemaRef;
       Lexicographic Lex;
-      MemberExpr *M;
+      Expr *LValue;
       bool Found;
 
     public:
-      FindMemberHelper(Sema &SemaRef, MemberExpr *M) :
+      FindLValueHelper(Sema &SemaRef, Expr *LValue) :
         SemaRef(SemaRef),
         Lex(Lexicographic(SemaRef.Context, nullptr)),
-        M(M),
+        LValue(LValue),
         Found(false) {}
 
       bool IsFound() { return Found; }
 
+      bool VisitDeclRefExpr(DeclRefExpr *E) {
+        DeclRefExpr *V = dyn_cast_or_null<DeclRefExpr>(LValue);
+        if (!V)
+          return true;
+        if (Lex.CompareExpr(V, E) == Lexicographic::Result::Equal)
```

Comment:
I am assuming we don't need to call `CompareExprSemantically` here?

Reply:
I don't believe we need to call `CompareExprSemantically` in order to compare two `DeclRefExprs`. We shouldn't need to create `PreorderASTs` for `DeclRefExprs` to compare them since the comparison will just involve unnecessary sorting of the `PreoderAST` nodes `LeafExprNode(0)` and `LeafExprNode(v)` for a `DeclRefExpr` `v`.